boot/bootutil/src/bootutil_misc.c

```diff
@@ -475,19 +475,34 @@ uint32_t bootutil_max_image_size(struct boot_loader_state *state, const struct f
     }
 
     return slot_trailer_off - trailer_padding;
-#elif defined(MCUBOOT_SWAP_USING_MOVE) || defined(MCUBOOT_SWAP_USING_OFFSET)
-    (void) state;
+#elif defined(MCUBOOT_SWAP_USING_MOVE)
+    (void) fap;
 
-    struct flash_sector sector;
-    /* get the last sector offset */
-    int rc = flash_area_get_sector(fap, boot_status_off(fap), &sector);
-    if (rc) {
-        BOOT_LOG_ERR("Unable to determine flash sector of the image trailer");
-        return 0; /* Returning of zero here should cause any check which uses
-                   * this value to fail.
-                   */
-    }
-    return flash_sector_get_off(&sector);
+    const struct flash_area *fap_pri = BOOT_IMG_AREA(state, BOOT_PRIMARY_SLOT);
+    assert(fap_pri != NULL);
+
+    size_t trailer_sz = boot_trailer_sz(BOOT_WRITE_SZ(state));
+    size_t sector_sz = boot_img_sector_size(state, BOOT_PRIMARY_SLOT, 0);
+    size_t padding_sz = sector_sz;
+
+    /* The trailer size needs to be sector-aligned */
+    trailer_sz = ALIGN_UP(trailer_sz, sector_sz);
+
+    return flash_area_get_size(fap_pri) - trailer_sz - padding_sz;
+#elif defined(MCUBOOT_SWAP_USING_OFFSET)
+    (void) fap;
+
+    const struct flash_area *fap_sec = BOOT_IMG_AREA(state, BOOT_SECONDARY_SLOT);
+    assert(fap_sec != NULL);
+
+    size_t trailer_sz = boot_trailer_sz(BOOT_WRITE_SZ(state));
+    size_t sector_sz = boot_img_sector_size(state, BOOT_SECONDARY_SLOT, 0);
+    size_t padding_sz = sector_sz;
+
+    /* The trailer size needs to be sector-aligned */
+    trailer_sz = ALIGN_UP(trailer_sz, sector_sz);
+
+    return flash_area_get_size(fap_sec) - trailer_sz - padding_sz;
 #elif defined(MCUBOOT_OVERWRITE_ONLY)
```

Comment:
If you ifdef around variable assigned with BOOT_PRIMARY_SLOT or BOOT_SECONDARY_SLOT, and passed where these are accepted,  then there is no difference between both blocks.

Reply:
@de-nordic You're right, I was anticipating https://github.com/mcu-tools/mcuboot/pull/2247, which will need different logic for swap-move and swap-offset. But considering only this PR, it makes indeed more sense to merge the two blocks. I updated my changes accordingly